### PR TITLE
docs: `paths` and `patterns` difference in `no-restricted-imports`

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
added difference between the `paths` and `patterns` option of `no-restricted-imports` rule.

#### Is there anything you'd like reviewers to focus on?
Fixes: #18253

### CODE REVIEW

File: docs/src/rules/no-restricted-imports.md [hunk 1]
```diff
@@ -286,6 +286,8 @@ import { AllowedObject } from "foo";
 
 This is also an object option whose value is an array. This option allows you to specify multiple modules to restrict using `gitignore`-style patterns.
 
+Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory.
```
[the remark this answers is not in the document]
Comment (Tanujkanti4441) [reply docs/src/rules/no-restricted-imports.md:289]: Done! added examples.

[the remark this answers is not in the document]
Comment (nzakas) [reply docs/src/rules/no-restricted-imports.md:289]:
```suggestion
Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory. For example:
```

File: docs/src/rules/no-restricted-imports.md [hunk 2]
```diff
@@ -286,6 +286,28 @@ import { AllowedObject } from "foo";
...
+
+```json
+"no-restricted-imports": ["error", {
+  "paths": [{
+    "name": "import-foo",
+  }]
+}]
+```
+
+Here rule will only restrict the `import-foo` module.
```
Comment (nzakas) [inline docs/src/rules/no-restricted-imports.md:299]:
```suggestion
This configuration restricts import of the `import-foo` module but wouldn't restrict the import of `import-foo/bar` or `import-foo/baz`. You can use `patterns` to restrict both:
```

File: docs/src/rules/no-restricted-imports.md [hunk 3]
```diff
@@ -286,6 +286,28 @@ import { AllowedObject } from "foo";
...
+
+```json
+"no-restricted-imports": ["error", {
+    "patterns": [{
+      "group": ["import1/*"],
+    }]
+}]
+```
+
+Here rule will restrict all modules inside `import1` directory.
```
Comment (nzakas) [inline docs/src/rules/no-restricted-imports.md:309]:
````suggestion
```json
"no-restricted-imports": ["error", {
    "paths": [{
      "name": "import-foo",
    }]
    "patterns": [{
      "group": ["import-foo/ba*"],
    }]
}]
```

This configuration restricts imports not just from `import-foo` using `path`, but also `import-foo/bar` and `import-foo/baz` using `patterns`.
````

**Review by nzakas — CHANGES_REQUESTED** [review]

File: docs/src/rules/no-restricted-imports.md (hunk 3 above)
Comment (Tanujkanti4441) [reply docs/src/rules/no-restricted-imports.md]: Done!

File: docs/src/rules/no-restricted-imports.md (hunk 2 above)
Comment (Tanujkanti4441) [reply docs/src/rules/no-restricted-imports.md]: Done!

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**